cmd/wazero/wazero.go

```diff
@@ -195,7 +201,13 @@ func doRun(args []string, stdOut io.Writer, stdErr logging.Writer, exit func(cod
 		rtc = wazero.NewRuntimeConfig()
 	}
 	if cache := maybeUseCacheDir(cacheDir, stdErr, exit); cache != nil {
-		rtc.WithCompilationCache(cache)
+		rtc = rtc.WithCompilationCache(cache)
+	}
+	if timeout != 0 {
+		newCtx, cancel := context.WithTimeout(ctx, timeout)
+		ctx = newCtx
+		defer cancel()
+		rtc = rtc.WithCloseOnContextDone(true)
```

Comment:
I assume returning a clone of `rtc` instead of mutating the receiver is a well-known pattern for Gophers, but it was a bit surprising for me :P